### PR TITLE
Updated to v1.31.0

### PR DESCRIPTION
**Pull request by tom-schier**
Updated the type definitions to align with version 1.31.0 version of the published standard

Admin:

- New defintions for PerformancMetrics scemas

Banking:

- some changes to existing scema documentation

Energy:

- EnergyAccountDetailV4
- EnergyAccountDetailResonseV4
- EnergyPlanContractV3
- EnergyPlanContractFullV3
- EnergyPlanControlledLoadV2
- EnergyPlanResponseV3
- EnergyPlanSolarFeedInTariffV3
- EnergyPlanTariffPeriodV2

### CODE REVIEW

File: .github/workflows/ghostbuster.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
       - uses: ./.github/actions/setup-for-scripts
+
```
Comment (sandersn) [inline .github/workflows/ghostbuster.yml:28]: please revert this change

File: types/consumer-data-standards/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# CDS Type Definition Library
```
Comment (sandersn) [inline types/consumer-data-standards/README.md:1]: The DT publisher ignores README.md and publishes its own. You can put this here but it won't be visible at (for example) https://www.npmjs.com/package/@types/aframe

File: types/consumer-data-standards/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Encountered an issue? We're here to help. Please visit our [issue reporting guidelines](https://d61cds.notion.site/Issue-Reporting-Guidelines-71a329a0658c4b69a232eab95822509b?pvs=4) for submitting an issue.
+
+## Stay Updated
+
+Join our newsletter to receive the latest updates, release notes, and alerts. [Subscribe here](https://consumerdatastandards.us18.list-manage.com/subscribe?u=fb3bcb1ec5662d9767ab3c414&id=a4414b3906).
+
+## License
+
+The artefact is released under the [MIT License](https://github.com/ConsumerDataStandardsAustralia/java-artefacts/blob/master/LICENSE), which allows the community to use and modify it freely.
```
Comment (sandersn) [inline types/consumer-data-standards/README.md:66]: specifically, the publisher-generated README already licenses types under the MIT, so this notice is redundant

File: types/consumer-data-standards/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    npm install @types/consumer-data-standards
+    ```
+    
+    Or
+    
+    ```bash
+    yarn install @types/consumer-data-standards
+    ```
+    
+2. **Configure TypeScript**: Ensure your `tsconfig.json` includes paths to where your types are stored:
```
Comment (sandersn) [inline types/consumer-data-standards/README.md:32]: I don't think this tsconfig configuration is needed. (at least in a typical tsconfig)

**Review by sandersn — CHANGES_REQUESTED** [review]
Type changes look fine to me but please revert the change to ghostbuster.yml.
Also, the new README isn't going to be published. The `@types` publisher generates its own README. The one on DT is intended for contributors to read, not for types consumers.

File: .github/workflows/ghostbuster.yml (hunk 1 above)
Comment (tom-schier) [reply .github/workflows/ghostbuster.yml]: Removed blank line

File: types/consumer-data-standards/README.md (hunk 4 above)
Comment (tom-schier) [reply types/consumer-data-standards/README.md]: Removed redundant sections

**Review by sandersn — APPROVED** [review]

**Comment by tom-schier** [comment]
Ready to merge
